### PR TITLE
Use processed inputs from command instantiation to write the command script.

### PR DESCRIPTION
**Pull request by mcovarr**
Changes are smaller than they appear, much of this is restructuring to do command instantiation and then flatMapping the rest of the logic using a half-built `InstantiatedCommand`. Slinging around a half-built `InstantiatedCommand` is IMHO a bit gross but preexisting.

This leverages the preprocessing and value mapping of inputs that takes place in command instantiation and captures those versions of the inputs. This enables the stdin redirection in particular to get the actual name of the standard input file as seen within the container.

### CODE REVIEW

File: supportedBackends/sfs/src/main/scala/cromwell/backend/impl/sfs/config/ConfigAsyncJobExecutionActor.scala [hunk 1]
```diff
@@ -100,7 +100,7 @@ sealed trait ConfigAsyncJobExecutionActor extends SharedFileSystemAsyncJobExecut
     val allInputs = providedWomInputs ++ optionalsForciblyInitializedToNone
     val womInstantiation = taskDefinition.instantiateCommand(allInputs, NoIoFunctionSet, identity, runtimeEnvironment)
 
-    val InstantiatedCommand(command, _, _, _) = womInstantiation.toTry.get
+    val InstantiatedCommand(command, _, _, _, _, _) = womInstantiation.toTry.get
```
Comment (kshakir) [inline supportedBackends/sfs/src/main/scala/cromwell/backend/impl/sfs/config/ConfigAsyncJobExecutionActor.scala]:
```scala
val command = womInstantiation.toTry.get.commandString
```

Comment (mcovarr) [reply supportedBackends/sfs/src/main/scala/cromwell/backend/impl/sfs/config/ConfigAsyncJobExecutionActor.scala]: I'm sure this made more sense at one point... 😦

File: backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala [hunk 2]
```diff
@@ -287,52 +289,59 @@ trait StandardAsyncExecutionActor extends AsyncBackendJobExecutionActor with Sta
...
     }
 
     val callable = jobDescriptor.taskCall.callable
+    def makeStringKeyedMap(list: List[(LocalName, WomValue)]): Map[String, WomValue] = list.toMap map { case (k, v) => k.value -> v }
 
-    val adHocFileCreations: ErrorOr[List[WomFile]] = callable.adHocFileCreation.toList.flatTraverse(
-      _.evaluateValue(commandInstantiationInputs, backendEngineFunctions).flatMap(validateAdHocFile)
-    )
+    val command = instantiatedCommandValidation flatMap { instantiatedCommand =>
+      val taskInputs = instantiatedCommand.taskInputs |> makeStringKeyedMap
```
Comment (Horneth) [inline backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala]: Don't you have access to the evaluated task inputs through the jobDescriptor already ? `jobDescritpor.evaluatedTaskInputs`

Comment (mcovarr) [reply backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala]: Evaluated inputs yes, preprocessed evaluated inputs no. I'll clean up this naming fail.

**Pull request merged**